aptos-move/framework/aptos-framework/sources/overloadable_fungible_asset.move

```diff
@@ -0,0 +1,169 @@
+        });
+    }
+
+    /// Withdraw `amount` of the fungible asset from `store` by the owner.
+    ///
+    /// The semantics of deposit will be governed by the function specified in OverloadFunctionStore.
+    public fun withdraw<T: key>(
+        owner: &signer,
+        store: Object<T>,
+        amount: u64,
+    ): FungibleAsset acquires OverloadFunctionStore {
+        let metadata_addr = object::object_address(&fungible_asset::store_metadata(store));
+        let owner_address = signer::address_of(owner);
+        assert!(exists<OverloadFunctionStore>(metadata_addr), error::not_found(EFUNCTION_STORE_NOT_FOUND));
+        let overloadable_store = borrow_global<OverloadFunctionStore>(metadata_addr);
+        dispatchable_withdraw(
+            owner_address,
+            store,
+            amount,
+            &overloadable_store.transfer_ref,
+            &overloadable_store.withdraw_function,
+        )
+    }
+
+    /// Deposit `amount` of the fungible asset to `store`.
+    ///
+    /// The semantics of deposit will be governed by the function specified in OverloadFunctionStore.
+    public fun deposit<T: key>(
+        store: Object<T>,
+        fa: FungibleAsset
+    ) acquires OverloadFunctionStore {
+        let metadata_addr = object::object_address(&fungible_asset::store_metadata(store));
+        assert!(exists<OverloadFunctionStore>(metadata_addr), error::not_found(EFUNCTION_STORE_NOT_FOUND));
+        let overloadable_store = borrow_global<OverloadFunctionStore>(metadata_addr);
+        dispatchable_deposit(
+            store,
+            fa,
+            &overloadable_store.transfer_ref,
+            &overloadable_store.deposit_function,
+        );
+    }
+
+    /// A transfer with a fixed amount debited from the sender
+    public fun transfer_fixed_send<T: key>(
+        _sender: &signer,
+        from: Object<T>,
+        to: Object<T>,
+        send_amount: u64,
+    ) acquires OverloadFunctionStore {
+        let store_address = object::object_address(&from);
+        assert!(exists<OverloadFunctionStore>(store_address), error::not_found(EFUNCTION_STORE_NOT_FOUND));
+        let overloadable_store = borrow_global<OverloadFunctionStore>(store_address);
+        let fa = fungible_asset::withdraw_with_ref(&overloadable_store.transfer_ref, from, send_amount);
+        deposit(to, fa);
+    }
+
+    /// A transfer with a fixed amount credited to the recipient
+    public fun transfer_fixed_receive<T: key>(
+        sender: &signer,
+        from: Object<T>,
+        to: Object<T>,
+        receive_amount: u64,
+    ) acquires OverloadFunctionStore {
+        let fa = withdraw(sender, from, receive_amount);
+        let store_address = object::object_address(&from);
+        let overloadable_store = borrow_global<OverloadFunctionStore>(store_address);
+        assert!(fungible_asset::amount(&fa) == receive_amount, error::aborted(EAMOUNT_MISMATCH));
```

Comment:
So here, the assumption is that once a hot potato FA has been withdrawn, the exact amount will necessarily end up in the receiver store

Correct?

Reply:
Yes because we use deposit_with_ref so it has no custom behavior.